### PR TITLE
upgrade project

### PR DESCRIPTION
**Pull request by ruy-dan**
This is a draft proposal to upgrade the project.
Dependencies have been upgraded to the latest version.
In accordance, some adjustments had to be made:
- Added phantom markers and small adjustments in lifetimes.
- Updated sqlx query returns.
- Added tokio-runtime sqlx feature

This upgraded version compiles and passes all tests.

### CODE REVIEW

**Comment by jamesjmeyer210** [comment]
Thank you @ruy-dan  for making this update. I'm reviewing your changes right now.

**Comment by ruy-dan** [comment]
> Thank you @ruy-dan  for making this update. I'm reviewing your changes right now.

Great. I hope it can be helpful to find a possible update for the project.

File: src/dao/db_context.rs [hunk 1]
```diff
@@ -1,67 +1,79 @@
...
 use std::sync::Arc;
 
 pub struct Table<'c, T>
 where
-    T: FromRow<'c, MySqlRow<'c>>,
+    T: FromRow<'c, MySqlRow>,
 {
     pub pool: Arc<MySqlPool>,
-    _from_row: fn(&MySqlRow<'c>) -> Result<T, sqlx::Error>,
+    _from_row: Box<dyn Fn(&'c MySqlRow) -> Result<T, sqlx::Error> + Send + Sync + 'c>,
```
Comment (jamesjmeyer210) [inline src/dao/db_context.rs:14]: Why does the callback `from_row` need to support `Send + Sync`? Is there something you're trying to support downstream? For example, I tried changing the `Table` definition to below and it builds and executes fine:
```rust
impl<'c, T> Table<'c, T>
where
    T: FromRow<'c, MySqlRow>,
{
    fn new(pool: Arc<MySqlPool>) -> Self {
        Table {
            pool,
            _from_row: T::from_row,
            _marker: PhantomData,
        }
    }
}
```
Same goes for `JoinTable`.

**Review by jamesjmeyer210 — COMMENTED** [review]
Once we resolve this comment, we get this PR through.

**Comment by ruy-dan** [comment]
I agree. I was thinking there might be scenarios where we could want to process rows in parallel or where the callback might be shared across threads, hence the Box and Send + Sync, but I agree that this brings unnecessary complexity for this context. I have updated the callback following your suggestion and pushed a new commit.

File: src/dao/db_context.rs (hunk 1 above)
Comment (ruy-dan) [reply src/dao/db_context.rs]: I agree. I was thinking there might be scenarios where we could want to process rows in parallel or where the callback might be shared across threads, hence the Box and Send + Sync, but I agree that this brings unnecessary complexity for this context. I have updated the callback following your suggestion and pushed a new commit.

**Review by jamesjmeyer210 — APPROVED** [review]
These changes should be sufficient for an upgrade.

**Pull request merged**